webapp/components/test-file-results.js

```diff
@@ -80,7 +83,65 @@ class TestFileResults extends TestRunsUIQuery(
   }
 
   static get observers() {
-    return ['fetchTestFile(path, testRuns)'];
+    return ['loadData(path, testRuns, structuredSearch)'];
+  }
+
+  async loadData(path, testRuns, structuredSearch) {
+    // Run a search query, including subtests, as well as fetching the results file.
+    let [searchResults, resultsTable] = await Promise.all([
+      this.fetchSearchResults(path, testRuns, structuredSearch),
+      this.fetchTestFile(path, testRuns),
+    ]);
+
+    if (resultsTable && searchResults) {
+      const test = searchResults.results.find(r => r.test === path);
+      if (test) {
+        let subtests = new Set(test.subtests);
+        let [first, ...others] = resultsTable;
+        const matches = others.filter(t => subtests.has(t.name));
```

Comment:
Just to confirm my understanding of this, it's the server that does the filtering, then we fetch the full results and filter that to match?